### ISSUE TITLE
wf-panel crashes after playing a song on spotify

### ISSUE THREAD

**Issue by dxcy47** [WayfireWM/wf-shell#283]
**Describe the bug**
whenever i open spotify and play a song, wf-panel crashes

**To Reproduce**
1. open spotify
2. play a song
3. wf-panel crashes
(note: it only crashes after playing a song. if i keep spotify open without playing anything it still works fine)

**Expected behavior**
expected wf-panel to work as normal without crashing after opening spotify

**Screenshots or stacktrace**

II 13-12-24 16:56:27.275 - [subprojects/wf-config/src/file.cpp:579] Reading XML configuration options from directory /opt/wayfire/share/wayfire/metadata/wf-shell
II 13-12-24 16:56:27.275 - [subprojects/wf-config/src/file.cpp:529] Reading XML configuration options from file /opt/wayfire/share/wayfire/metadata/wf-shell/background.xml
II 13-12-24 16:56:27.275 - [subprojects/wf-config/src/file.cpp:529] Reading XML configuration options from file /opt/wayfire/share/wayfire/metadata/wf-shell/dock.xml
II 13-12-24 16:56:27.275 - [subprojects/wf-config/src/file.cpp:529] Reading XML configuration options from file /opt/wayfire/share/wayfire/metadata/wf-shell/panel.xml
EE 13-12-24 16:56:27.276 - [subprojects/wf-config/src/file.cpp:341] Error in file /home/decoy/.config/wf-shell.ini:100, invalid option value!
EE 13-12-24 16:56:27.276 - [subprojects/wf-config/src/file.cpp:341] Error in file /home/decoy/.config/wf-shell.ini:100, invalid option value!
Error loading file: /usr/share/icons/gnome/48x48/categories/preferences-system.png
target button 508
set width 192

** (wf-panel:2594): ERROR **: 16:56:36.190:
unhandled exception (type std::exception) in signal handler:
what: map::at

fish: Job 1, 'wf-panel' terminated by signal SIGTRAP (Trace or breakpoint trap)

-----
this is the whole log from starting wf-panel to it crashing

**Wayfire version**
wayfire-git 0.9.0 installed with wf-install

**Comment by NamorNiradnug** [comment]
Hi, could you please check whether #284 fixes the issue?

**Comment by bell07** [comment]
Confirm the patch in #284 avoid the panel crash if spotify play is started first time.  In addition it solves my issue the keepassxc tray icon was not shown.

But now spotify freezes for ~30 seconds if play is started first time. The music is playing, but cannot be controlled, paused or window be closed. After ~30 seconds (time measured with timer) the spotify gets useable again.
Still have to disable the notifications in spotify to avoid this freeze.

**Comment by NamorNiradnug** [comment]
> In addition it solves my issue the keepassxc tray icon was not shown.

Thew patch is not related to the tray widget at all, probably you done something else in order to fix it.

> But now spotify freezes for ~30 seconds if play is started first time. The music is playing, but cannot be controlled, paused or window be closed. After ~30 seconds (time measured with timer) the spotify gets useable again.

Hmm, interesting... Sounds like a Spofify's issue to me. Do I get it right:
- you open Spotify
- start playing some music
- the music is playing, the notification is there, but Spotify's UI is freezed
- then it unfreezes and playing next tracks it fine
?

**Comment by bell07** [comment]
You are right, the keepassxc issue is not solved yet. After reboot the tray icon is away again...

- I open Spotify
- I start paying music
- The Notification does NOT appear
- Before the patch the Panel was crashed at this point
- After the patch I get a freeze in spotify. 
- Freeze means: The time remains at 0:00 and the "play" button is not swapped to "pause" button. I cannot stop playing
- After ~30 seconds freeze the time jumps to 0:30 and spotify gets usable again

spotify --debug-level=MAX does not provide anything useful additional informations. I get only the two lines on start:
```
Neither gnome-integration-spotify nor spotify-tray are installed.
Launching spotify without systray integration.
```
contradicting this message I get spotify tray icon...

I disabled the notifications in wf-panel and started dunst
- The notification appears. (Music cover image at the left and Album name + Track name in two lines at the right)
- No freeze
- No change in spotify --debug-level=MAX output

Tried https://github.com/emersion/mako too, it does work too

**Comment by soreau** [comment]
> You are right, the keepassxc issue is not solved yet. After reboot the tray icon is away again...

For what it's worth, there is an option in KeepassXC to display the tray icon or not, which seems to work fine here.

**Comment by bell07** [comment]
> > You are right, the keepassxc issue is not solved yet. After reboot the tray icon is away again...
> 
> For what it's worth, there is an option in KeepassXC to display the tray icon or not, which seems to work fine here.

This Issue I solved by `autostart1 = sleep 5; keepassxc` now.

Seems to be a timing issue in autostart processing.

**Comment by NamorNiradnug** [comment]
I've found another bug and it should work now (at least I've finally installed spotify and it works for me). Although it looks ugly because notification's text and image are side-by-side and image is large. Probably the best solution is to scale it (gonne fix it soon).

**Comment by bell07** [comment]
Confirm the spotify notification does work for me as expected and there is no freeze
